spyderlib/spyder.py

```diff
@@ -2735,6 +2754,7 @@ def restart(self):
         env['SPYDER_ARGS'] = spyder_args
         env['SPYDER_PID'] = str(pid)
         env['SPYDER_IS_BOOTSTRAP'] = str(is_bootstrap)
+        env['SPYDER_RESET'] = str(reset)

```

Comment:
So the problem was only with the reset then,, or even if you just try to restart spyder it was failing?